app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx

```diff
@@ -0,0 +1,175 @@
+    { id: 411, name: 'Stepper', parentId: 4 },
+    { id: 412, name: 'In-form Tables', parentId: 4 },
+    { id: 413, name: 'User card', parentId: 4 },
+    { id: 414, name: 'Top-level navigation', parentId: 4 },
+    { id: 501, name: 'Page layout components', parentId: 5 },
+    { id: 502, name: 'Master-detail', parentId: 5 },
+    { id: 503, name: 'Full-screen table', parentId: 5 },
+    { id: 504, name: 'Full-screen form', parentId: 5 },
+    { id: 505, name: 'Dashboard', parentId: 5 },
+    { id: 506, name: 'Catalog', parentId: 5 },
+    { id: 601, name: 'Home', parentId: 6 },
+    { id: 602, name: 'Catalog', parentId: 6 },
+    { id: 603, name: 'Product Details', parentId: 6 },
+    { id: 604, name: 'Favorites', parentId: 6 },
+    { id: 605, name: 'Comparison', parentId: 6 },
+    { id: 606, name: 'Check-out form', parentId: 6 },
+    { id: 60701, name: 'Products List', parentId: 607 },
+    { id: 60702, name: 'Product Form', parentId: 607 },
+    { id: 60703, name: 'Sales report', parentId: 607 },
+    { id: 60704, name: 'Marketing dashboard', parentId: 607 },
+    { id: 60705, name: 'Categories list editor', parentId: 607 },
+    { id: 201, name: 'Authentication', parentId: 2 },
+    { id: 202, name: 'Integration with API', parentId: 2 },
+    { id: 203, name: 'Routing', parentId: 2 },
+    { id: 204, name: 'Localization', parentId: 2 },
+    { id: 2, name: 'Shared services' },
+    { id: 3, name: 'UUI Customization' },
+    { id: 4, name: 'Shared Components' },
+    { id: 5, name: 'Pages Template Components' },
+    { id: 6, name: 'Pages' },
+    { id: 607, name: 'Admin area', parentId: 6 },
+];
+
+export default function CitiesTable() {
+    const [listState, setListState] = useState<DataSourceState>({
+        visibleCount: 20, // how many items to load initially?
+    });
+
+    // Create DataSource instance for your table.
+    // For more details go to the DataSources example
+    const tasksDs = useArrayDataSource<Task, number, unknown>(
+        { 
+            items: tasks,
+            isFoldedByDefault: () => false,
+            getParentId: ({ parentId }) => parentId,
+        },
+        [],
+    );
+
+    // Create View according to your tableState and options
+    const view = tasksDs.useView(listState, setListState, {
+        getRowOptions: useCallback(
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            (item) => ({ isSelectable: true }), // can set row options here
+            [],
+        ),
+    });
+    const rows = view.getVisibleRows().map((row) => {
+        let content: React.ReactNode;
+        if (row.isLoading) {
+            content = <TextPlaceholder />;
+        } else {
+            content = (
+                <Text fontWeight="600" fontSize="16">
+                    {row.value.name}
+                </Text>
+            );
+        }
+        
+        const getIndent = () => {
+            return 12 + (row.indent - 1) * 24;
```

Comment:
If 12 it's a global padding for all items, let's set it via css and exclude from this calculation